packages/rest-crud/README.md

```diff
@@ -0,0 +1,65 @@
+# @loopback/rest-crud
+
+REST API controller implementing default CRUD semantics.
+
+## Overview
+
+This module allows applications to quickly expose a model via REST API without
+having to implement a custom controller class.
+
+## Installation
+
+```sh
+npm install --save @loopback/rest-crud
+```
+
+## Basic use
+
+1. Define your model class, e.g. using `lb4 model` tool.
+
+2. Create a Repository class, e.g. using `lb4 repository` tool.
+
+3. Create a base REST CRUD controller class for your model.
+
+   ```ts
+   const CrudRestController = defineCrudRestController<
+     Product,
+     typeof Product.prototype.id,
+     'id'
+   >(Product, {basePath: '/products'});
+   ```
+
+4. Create a new subclass of the base controller class to configure repository
+   injection.
+
+   ```ts
+   class ProductController extends CrudRestController {
+     constructor(@repository(ProductRepository) repo: ProductRepository) {
```

Comment:
If we skip the repository, can we inject a datasource instead?

Reply:
Yes, I believe injecting a datasource should be a viable option too. The actual implementation does not prescribe any IoC setup, the base controller class is expecting to receive a repository instance. It's up to the code extending the controller class to decide how to obtain the repository.

I believe the following solution should work as an alternative approach too:

```ts
class ProductController extends CrudRestController {
  constructor(
    @inject('datasources.db') dataSource: juggler.DataSource,
  ) {
    const repo = new DefaultCrudRepository<
      Product,
      typeof Product.prototype.id
    >(Product, db);

    super(repo);
  }
}
```

@raymondfeng Would you like me to open a new issue to add a test & documentation for this approach?